src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateVersions = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(stateCopy, extraData);
+        break;
+      case 'removeProperties':
+        removeProperties(stateCopy, keysToRemove);
+        break;
+      default:
+        clear(stateCopy);
+    }
+    stateVersions.push(stateCopy);
+    stateCopy = { ...stateCopy };
```

Comment:
push to `stateVersions` copy of stateCopy and you can remove this line of code
```stateCopy = { ...stateCopy };```
```suggestion
    stateVersions.push({ ...stateCopy });
```